client/components/domains/registrant-extra-info/ca-form.jsx

```diff
@@ -109,81 +114,86 @@ export class RegistrantExtraInfoCaForm extends React.PureComponent {
 		} );
 	}
 
-	handleChangeEvent = event => {
-		const { target } = event;
-		let value = target.value;
+	validateContactDetails = contactDetails => {
+		wpcom.validateDomainContactInformation(
+			contactDetails,
+			this.props.getDomainNames(),
+			( error, data ) => {
+				this.setState( {
+					errorMessages: ( data && data.messages ) || {},
+				} );
+			}
+		);
+	};
 
-		if ( target.type === 'checkbox' ) {
-			value = target.checked;
+	handleChangeEvent = event => {
+		const { value, name, checked, type, id } = event.target;
+		const newContactDetails = {};
+
+		if ( this.props.contactDetails.hasOwnProperty( name ) ) {
```

Comment:
It'd be nice to add a few more hints about what's going on here - the organization field lives in the top level `contactDetails` rather than extra, and it needs back end validation, but it just happens to have those two properties. 

In the same vein, I'd rather call `organization` out explicitly than check for properties, mostly just out of paranoia. It's not like we'll get a collision in .CA or a `contactDetails` object without an `organization` property, but I've been unpleasantly surprised a couple of times.